src/main/java/com/fasterxml/jackson/databind/introspect/JacksonAnnotationIntrospector.java

```diff
@@ -240,6 +240,45 @@ public String[] findEnumValues(Class<?> enumType, Enum<?>[] enumValues, String[]
         return names;
     }
 
+    @Override // since 2.15
+    public String[] findEnumValues(MapperConfig<?> config, Class<?> enumType, Enum<?>[] enumValues, String[] names, AnnotatedClass annotatedClass) {
+        // First collect all JsonProperty.value()
+        HashMap<String, String> enumToPropertyMap = new HashMap<>();
+        for (AnnotatedField field : annotatedClass.fields()) {
+            JsonProperty property = field.getAnnotation(JsonProperty.class);
```

Comment:
One problem here is that since we are not calling method that looks for rename, this only works for `@JsonProperty` but not -- for example -- equivalent JAXB annotation.

Reply:
... but I now realize this is how code already was. So not a new deviation. And is exactly why mix-ins won't work because, yeah.

Reply:
Sure: method `findNameForDeserialization(Annotated)` is used for Fields, Methods, constructor parameters.